.github/workflows/check-code-quality.yml

```diff
@@ -31,3 +31,10 @@ jobs:
       uses: ludeeus/action-shellcheck@master
       with:
         severity: style
+
+    - name: Check for broken links
+      run: | 
+        find . -type f \( -name "*.txt" -o -name "*.md" -o -name "*.py" -o -name "*.rst" \) | 
+          xargs grep -Eo "(http|https)://[a-zA-Z0-9./?=_%:-]*" | 
+          sed 's/:/;/' | 
+          xargs -n 1 -P 8 ./.github/workflows/check-url.sh
```

Comment:
Neat, it works! Though, some observations:

- The grep pattern I used will include trailing periods, which are somewhat common for links that are included as part of a sentence.
- We should intentionally exclude the `CHANGES.md` file, since it has a _ton_ of GitHub PR links, and this will trigger an Error 429 - Too Many Requests.
- The output is really noisy when including links that are OK.

Reply:
For the noisy output: maybe the _numbers_ of good/bad/total links, and the _list_ of bad links (with locations?) would be a good amount of output?

Reply:
By numbers, do you mean a summary similar to pytest output?

![image](https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/16181459/515f50ea-30d8-4ed9-b364-8d030f34be37)

Reply:
Exactly!